STYLE.md

```diff
@@ -0,0 +1,91 @@
+### Code formatting
+
+To format your code, run `go fmt`.
+
+> Note: Your editor may do this automatically.
+
+
+### Imports
+
+Import statements are grouped into 3 sections: standard library, 3rd party
+libraries, juju imports. The tool "go fmt" can be used to ensure each
+group is alphabetically sorted. eg:
+
+```go
+    import (
+        "fmt"
+        "time"
+
+        "labix.org/v2/mgo"
+        "github.com/juju/loggo"
+        "github.com/juju/worker/v3"
+        gc "gopkg.in/check.v1"
+
+        "github.com/juju/juju/state"
+    )
+```
+
+### Method documentation
+
+Do not cut and paste swabs of text from the interface method declaration to each and every implementation method.
+Just use doc comment references.
```

Comment:
This is the opposite of what we're been encouraging informally. I don't think a reference to the interface is very useful. This feels like a cop-out

Also, this is probably the wrong way around in most cases. The way we use interfaces is mostly for dependency inversion, so we end up with many interfaces shadowing a small number of concrete implementations. To have the interfaces and not the underlying concrete implementation seems silly, and contradicts the next subheading

Reply:
Each exported method should have a comment header. Saying it implements interface blah isn't useful. We need to comment the implementation details which are important to know.

Reply:
> This is the opposite of what we're been doing and encouraging.
> 
> I get that most IDEs can navigate straight to the interface, and that you want to avoid repetition, which is fair enough, but it feels like a cop out to just say "check out the interface" instead of documenting

We discussed this in the x-team and also I put up a Google Doc where people could provide input and in that doc gave the rationale. There was broad agreement we want to do this. It is far worse to try and maintain large chunks of text which can easily diverge and become incorrect over time.